safir/src/safir/metrics/_testing.py

```diff
@@ -0,0 +1,235 @@
+    ``model_dump(mode="json")`` serialization of the models for comparison.
+
+    """
+
+    def assert_published(
+        self,
+        expected: ModelDumpList,
+        *,
+        any_order: bool = False,
+    ) -> None:
+        """Assert that all of the expected payloads were published.
+
+
+        Parameters
+        ----------
+        expected
+            A list of expected event payload dicts, to be compared with the
+            ``model_dump(format="json")`` serialization of the actual published
+            payloads.
+        any_order
+            If true, then the expected payload list must be an ordered subset
+            of the actual published payloads. This is like the ``any_order``
+            parameter in `unittest.mock.Mock` methods.
+
+        Raises
+        ------
+        PublishedTooFewError
+            Expected more events than have actually been published
+        NotPublishedError
+            Some expected items were not published
+        NotPublishedConsecutivelyError
+            Expected events were not published consecutively
+        """
+        if len(expected) > len(self):
+            actual = [model.model_dump(mode="json") for model in self]
+            raise PublishedTooFewError(expected, actual, self)
+
+        if any_order:
+            self._unordered(expected)
+        else:
+            # Try unordered first to differentiate between the cases where some
+            # expected events were not published at all, vs all of the expected
+            # events were published, just not in the expected order.
+            self._unordered(expected)
+            self._ordered(expected)
+
+    def assert_published_all(
+        self,
+        expected: ModelDumpList,
+        *,
+        any_order: bool = False,
+    ) -> None:
+        """Assert that all of the expected payloads were published, and only
+        the expected payloads, were published.
+
+
+        Parameters
+        ----------
+        expected
+            A list of expected event payload dicts, to be compared with the
+            ``model_dump(format="json")`` serialization of the actual published
+            payloads.
+        any_order
+            If true, then the expected payloads must be in the same order as
+            the actual published payloads. This is like the ``any_order``
+            parameter in `unittest.mock.Mock` methods.
+
+        Raises
+        ------
+        NotPublishedError
+            Some expected items were not published
+        NotPublishedConsecutivelyError
+            Expected events were not published consecutively
+        PublishedCountError
+            A different number of events were published than were expected
+        """
+        if len(expected) != len(self):
+            actual = [model.model_dump(mode="json") for model in self]
+            raise PublishedCountError(expected, actual, self)
+        self.assert_published(expected, any_order=any_order)
+
+    def _unordered(
```

Comment:
Maybe `_check_unordered`? Right now the method name is grammatically a noun, which implies that it's returning some piece of data, but it's actually performing an action, so it feels like it should be a verb.